src/components/comments/CommentsPanelListing.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import CommentsPanelListingItem from './CommentsPanelListingItem';
+
+export const CommentsPanelListing = (props) => {
+  const { comments } = props;
+  return (
+    <div>
+      {comments &&
+        comments.length > 0 &&
+        comments.map((data, index) => (
+          <CommentsPanelListingItem key={index} data={data} />
+        ))}
+    </div>
+  );
+};
+
+const mapStateToProps = (state) => {
+  const { data } = state.commentsPanel;
+
+  return {
+    comments: data,
+  };
+};
+
+CommentsPanelListing.propTypes = {
+  comments: PropTypes.array,
+};
+
+export default connect(mapStateToProps)(CommentsPanelListing);
```

Comment:
Just thinking out loud - wouldn't connecting just the parent comments panel be more efficient ? One `connect` that fetches data and decorates the action creators instead of two ?

Reply:
It won't in this case. You have to think also at renderings involved.  You see CommentsPanel re-renders only when windowId or docId change. If I were to do that would trigger re-renderings when props change(from connect).  I only want to re-render the CommentsPanelListing when the data within it changes.